File: DataRetention/app/src/main/java/com/melihakkose/dataretention/MainActivity.java
package com.melihakkose.dataretention;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    TextView textView;
    SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editText =findViewById(R.id.editTextNumber);
        textView=findViewById(R.id.textView);

        sharedPreferences =this.getSharedPreferences("com.melihakkose.dataretention", Context.MODE_PRIVATE);
        int age =40;

        int storedAge=sharedPreferences.getInt("storedAge",0);
        if(storedAge==0){
            textView.setText("Your Age:");
        }else{
            textView.setText("Your Age:"+storedAge);
        }


    }
    public void save(View view) {
        if(!editText.getText().toString().matches("")){
            int age=Integer.parseInt(editText.getText().toString());
            textView.setText("Your age:"+age);
            sharedPreferences.edit().putInt("storedAge",age).apply();
        }
    }
    public void delete(View view) {
        int storedData=sharedPreferences.getInt("storedAge",0);
        if(storedData!=0){
            sharedPreferences.edit().remove("storedAge").apply();
            textView.setText("Your Age:");
        }

    }
}

File: StoringData/app/src/main/java/com/melihakkose/storingdata/MainActivity.kt
package com.melihakkose.storingdata

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    //Lateinit sonra initialize edicem demek
    lateinit var sharedPreferences:SharedPreferences

    var agePreferences:Int?=null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //INITIALIZE
         sharedPreferences= this.getSharedPreferences("com.melihakkose.storingdata",
            Context.MODE_PRIVATE)

          agePreferences=sharedPreferences.getInt("myAge",0)

        if(agePreferences==0){
            textView.text="Your Age:"
        }else{
            textView.text="Your Age: $agePreferences "
        }

    }

    fun save(view: View){

        //SharedPreferences -> Kucuk veriler kaydedilirken kullanilir. (Cok az veri ile hizli calisir)
        //MODE_PRIVATE olusturulan uygulama icerisinden cagiriliyor



        val age=editTextAge.text.toString().toIntOrNull()
        if(age!=null){
            textView.text="Your Age: "+ age
            sharedPreferences.edit().putInt("myAge",age).apply()
        }else{
            textView.text="Your Age: Required!"
        }

    }

    fun delete(view:View){
        agePreferences=sharedPreferences.getInt("myAge",0)

        if(agePreferences!=0){
            sharedPreferences.edit().remove("myAge").apply()
            textView.text="Your Age: "
        }

    }
}